File: fastframe/src/main/java/com/code/fastframe/baseactivity/BaseStatuBarActivity.java
package com.code.fastframe.baseactivity;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import com.code.fastframe.R;
import com.code.fastframe.ciface.IBaseStatuBarActivity;
import com.code.fastframe.utils.StatusBarUtil;

public abstract class BaseStatuBarActivity extends AppCompatActivity
    implements IBaseStatuBarActivity {

  @Override
  protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
  }

  @Override
  protected void onResume() {
    super.onResume();
    if (needNavigationBarTransparent()) {
      //            StatusBarUtil.setTransparent(this);
    }
    if (needStatuBarTransparent()) {
      StatusBarUtil.setColor(this, getResources().getColor(R.color.titleBarColor), 0);
    }
  }

  @Override
  public boolean needStatuBarTransparent() {
    return false;
  }

  @Override
  public boolean needNavigationBarTransparent() {
    return false;
  }
}

File: fastframe/src/main/java/com/code/fastframe/baseactivity/BaseStatuBarActivity.kt
package com.code.fastframe.baseactivity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.code.fastframe.R.color
import com.code.fastframe.ciface.IBaseStatuBarActivity
import com.code.fastframe.utils.StatusBarUtil

abstract class BaseStatuBarActivity : AppCompatActivity(), IBaseStatuBarActivity {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
  }

  override fun onResume() {
    super.onResume()
    if (needNavigationBarTransparent()) {
      //            StatusBarUtil.setTransparent(this);
    }
    if (needStatuBarTransparent()) {
      StatusBarUtil.setColor(this, resources.getColor(color.titleBarColor), 0)
    }
  }

  override fun needStatuBarTransparent(): Boolean {
    return false
  }

  override fun needNavigationBarTransparent(): Boolean {
    return false
  }
}